include/AutomatableModelView.h

```diff
@@ -97,10 +102,28 @@ class AutomatableModelViewSlots : public QObject
 public slots:
 	void execConnectionDialog();
 	void removeConnection();
+	void addSongAutomationNode();
+	void addSongAutomationNodeAndClip();
+	void updateSongNearestAutomationNode();
+	void removeSongNearestAutomationNode();
 	void editSongGlobalAutomation();
 	void unlinkAllModels();
 	void removeSongGlobalAutomation();
 
+private:
+	// gets the automationTrack with the most amount of clips connected to it
+	// if this track doesn't exists and "canAddNewTrack", then add a new one else return nullptr
+	// "clips" = clips that are connected to this model
+	AutomationTrack* getCurrentAutomationTrack(std::vector<AutomationClip*>* clips, bool canAddNewTrack);
+	// gets the clip that start before or after the song time position (playback pos)
+	// if this clip doesn't exists and "canAddNewClip", then add a new one else return nullptr
+	AutomationClip* getCurrentAutomationClip(AutomationTrack* track, bool canAddNewClip, bool searchAfter);
+	// gets the automationNode closest to the song time position (playback pos)
+	// "clipOut" is the clip that has the node
+	// can return nullptr on "clipOut"
+	const TimePos getNearestAutomationNode(AutomationTrack* track, AutomationClip** clipOut);
```

Comment:
`AutomationClip**` looks very "C-like". Is this due to the client which uses this method or did you make the decision to use this construct? `getCurrentAutomationTrack` uses a vector of pointers which would be more "C++-style".

Reply:
`clipOut` is an automation clip pointer output. I can make the function return an `std::pair` if you like that better,

Reply:
Personally I am not a fan of `std::pair` returns either because they do not really communicate well what's being returned. Alternatively you could consider to define a small struct which is returned by this method, e.g.:
```
struct AutomationClipAtTimePos
{
    TimePos timePos;
    AutomationClip* clip;
}
```

This way it would also be clear at the calling side what data is being used, e.g.
```
const AutomationClipAtTimePos clipInfo = getNearestAutomationNode(track);

clipInfo.timePos // do something with the position
clipInfo.clip // do something with the clip
```